### PR TITLE
Certificates Self-Signed

### PR DESCRIPTION
**Pull request by guiadco**
Add the role certs and the modification for integration Self-Signed Certificates for AllSpark Release.

### CODE REVIEW

File: install.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@
   - name: Setup system dependencies
     include_role:
       name: system
+  - name: Setup Certificats
```
Comment (Xide) [inline install.yml]: `s/Certificats/Certificates`

File: roles/certs/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+
+- name: Create traefik data certs folder
```
Comment (Xide) [inline roles/certs/tasks/main.yml]: Certificates should be in the `{{ allspark_root_directory }}/data/secrets/` directory

File: roles/certs/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  file:
+    state: directory
+    path: "{{ allspark_root_directory }}/data/traefik/certs"
+
+- name: create self-signed SSL cert
+  command: openssl req -new -nodes -x509 -subj "/C=FR/ST=Haute-Garonne/L=Toulouse/O=CRE/CN=${ansible_fqdn}" -days 3650 -keyout "{{ allspark_root_directory }}/data/traefik/certs/traefik.key" -out "{{ allspark_root_directory }}/data/traefik/certs/traefik.crt" arg1
```
Comment (Xide) [inline roles/certs/tasks/main.yml]: Specific certificates informations to remove / put in configuration: ```-subj "/C=FR/ST=Haute-Garonne/L=Toulouse/O=CRE/CN=${ansible_fqdn}"```

+ What is this `arg1` at the end of the command ?
+ `${ansible_fqdn}` must be replaced by `{{ allspark_root_directory }}` as FQDN is not configured to reflect the allspark configuration.

File: roles/certs/tasks/main.yml (hunk 2 above)
Comment (Xide) [reply roles/certs/tasks/main.yml]: And missing the `{{ allspark_root_directory }}/config/traefik` used below

File: roles/network/tasks/main.yml [hunk 4]
```diff
@@ -18,9 +18,12 @@
     image: "{{ allspark_traefik.image }}"
     volumes:
       - "{{ allspark_root_directory }}/config/traefik/traefik.toml:/etc/traefik/traefik.toml"
+      - "{{ allspark_root_directory }}/data/traefik/certs/traefik.crt:/certs/traefik.crt"
```
Comment (Xide) [inline roles/network/tasks/main.yml]: Incorrect spacing : useless newlines

**Comment by Xide** [comment]
My bad, this is not approved, i was trying to edit the review.
There are still some issues with the PR for now : 
- There is still specific informations in the certificate generation task : `/C=FR/ST=Haute-Garonne/L=Toulouse/O=CRE`
- By default, self signed certificates should be disabled, and triggered by a configuration change in the `group_vars`.

Thanks

File: roles/network/tasks/main.yml [hunk 5]
```diff
@@ -2,6 +2,16 @@
   docker_network:
     name: "allspark"
 
+- name: Create traefik data certs folder
+  file:
+    state: directory
+    path: "{{ allspark_root_directory }}/data/secrets/traefik/certs"
+
+- name: create self-signed SSL cert
+  command: openssl req -new -nodes -x509 -subj "/C=FR/ST=Haute-Garonne/L=Toulouse/O=CRE/CN={{ allspark_root_domain }}" -days 3650 -keyout "{{ allspark_root_directory }}/data/secrets/traefik/certs/traefik.key" -out "{{ allspark_root_directory }}/data/secrets/traefik/certs/traefik.crt"
```
Comment (Xide) [inline roles/network/tasks/main.yml]: There is still specific informations in the certificate generation task : `/C=FR/ST=Haute-Garonne/L=Toulouse/O=CRE`

**Comment by guiadco** [comment]
Certificates and Variables Validated

Need to validate the PR

File: roles/network/tasks/main.yml [hunk 6]
```diff
@@ -7,6 +7,48 @@
...
+    force: true
+  when: allspark_traefik.selfsigned
+
+- name: Generate a Self Signed OpenSSL certificate
+  openssl_certificate:
+    path: "{{ allspark_root_directory }}/data/secrets/traefik/certs/traefik.crt"
+    privatekey_path: "{{ allspark_root_directory }}/data/secrets/traefik/certs/private/traefik.pem"
+    csr_path: "{{ allspark_root_directory }}/data/secrets/traefik/certs/traefik.csr"
+    provider: selfsigned
+    force: true
```
Comment (Xide) [inline roles/network/tasks/main.yml]: Force here break indepotency, from the docs : `Generate the certificate, even if it already exists.`

**Comment by guiadco** [comment]
Waiting for your advise :-)

File: group_vars/all.yml [hunk 7]
```diff
@@ -4,12 +4,7 @@
 allspark_root_domain: allspark.localhost
 allspark_root_directory: /opt/allspark
 allspark_admin_user: admin
-allspark_country: FR
-allspark_state: Haute-Garonne
-allspark_location: Toulouse
-allspark_organisation: CRE
-organizational_unit: TheFkinCompany
-allspark_common_name: "*.{{ allspark_root_domain }}"
+# For example country: FR
```
Comment (Xide) [inline group_vars/all.yml]: Misleading comment location, should be in the `allspark_traefik.selfsigned` section.

File: group_vars/all.yml [hunk 8]
```diff
@@ -22,7 +17,14 @@ allspark_monitoring:
   enabled: true
 
 allspark_traefik:
-  selfsigned: false
+  selfsigned: 
+    enable: false
```
Comment (Xide) [inline group_vars/all.yml]: Should be called `enabled` by convention. See the `kubespray` repository for [an applied example](https://github.com/kubernetes-incubator/kubespray/blob/e24f888bc44386f2b86b300c097a24cfd02c1ea5/inventory/sample/group_vars/k8s-cluster.yml#L172)

**Comment by guiadco** [comment]
I make the correction, waiting for your rereading and approval.

**Pull request merged**